lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -125,14 +115,6 @@ class AwsCompileS3Events {
               bucketName = bucketRef;
             }
 
-            if (/[A-Z]/.test(bucketName)) {
```

Comment:
This validation should I guess stay in, but it'll be good to use regex from definition for it

Reply:
I would need to export the pattern from `awsProvider` in some way. I don't want to mess with `module.exports`, and a class member seems not ideal semantically. Maybe we should put the pattern in some common module?

Reply:
I think this pattern is accessible on config schema, so we can probably get to it via `this.serverless.configSchema.definitions.awsS3BucketName.pattern`

Reply:
Yeah that was my first thought, but when running the tests I see that `this.serverless.configSchema` is undefined. Maybe it's because the tests aren't using the `runServerless` test util?

Reply:
> Maybe it's because the tests aren't using the runServerless test util?

Yes, that's the problem with tests (and the reason we write all new ones with `runServerless`). Ideally those tests should be refactored to `runServerless`, but ofc that's not in a scope of this PR.

So for that you may ensure that mocked serverless intance, has this object.